Csproj settings fix

jyardin opened the pull request:
When the solution file path is not  above the project path, the method `GetConfigPathOutsideSolution` is used and the search stops right before the project directory (which is not what is expected).  
Ex:  
For this code hierarchy:
 * solutionDir
   * MyProject.sln
 * projectDir
   * MyProject.csproj
   * Settings.XamlStyler
   * Views
     * MyView.xaml

The file Settings.XamlStyler is not found, because the method `GetConfigPathOutsideSolution` stops at the "View" directory.
 
To solve this, I replaced the `GetConfigPathOutsideSolution` call with a `GetConfigPathInsideSolution` call (renamed `GetConfigPathBetweenPaths`) with the project path as an argument.

Additionally, I added a lookup inside the project file references so the project can reference the Settings.XamlStyler file outside its "physical" directory.

NicoVermeir commented:
@jyardin thanks for contributing!

I see that you made this change in V2, can you move it over to the XamlStyler3 package?
V2 is in maintenance mode, all new work is happening in V3

thanks again!

jyardin commented:
@NicoVermeir I added the fix to XamlStyler3 in https://github.com/Xavalon/XamlStyler/pull/102/commits/af60584273dfcd4029eac7a88a86bd9d80782253.
For information, the project files lookup I added has a lower priority than the directory lookup.
